src/Language/PureScript/Errors.hs

```diff
@@ -733,10 +737,21 @@ prettyPrintSingleError (PPEOptions codeColor full level showWiki) e = flip evalS
       paras $ [ line "Wildcard type definition has the inferred type "
               , markCodeBox $ indent $ typeAsBox ty
               ] ++ renderContext ctx
-    renderSimpleErrorMessage (HoleInferredType name ty ctx) =
-      paras $ [ line $ "Hole '" ++ markCode name ++ "' has the inferred type "
-              , markCodeBox $ indent $ typeAsBox ty
-              ] ++ renderContext ctx
+    renderSimpleErrorMessage (HoleInferredType _ _ _ (TSBefore _)) =
+      internalError "oh noez"
+    renderSimpleErrorMessage (HoleInferredType name ty ctx typeSearch) =
+      case typeSearch of
+        TSBefore _ -> internalError "type search should've been run at this point"
+        TSAfter idents ->
+          paras $ [ line $ "Hole '" ++ markCode name ++ "' has the inferred type "
+                  , markCodeBox $ indent $ typeAsBox ty
+                  ]
+          ++ [line "You could substitute the hole with one of these values:"]
+          ++ map (\(i, t) -> (Box.text (showQualified runIdent i))
```

Comment:
Please indent these lines with `indent` and use `markCodeBox` too.

It would be great if the `::`s could be aligned too. A simple way would be to `unzip` the list of tuples, `vcat` each side and combine the two columns with `<>`.